### PR TITLE
Add examples to lowest common ancestors algorithms

### PR DESCRIPTION
**Pull request by dtekinoglu**

### CODE REVIEW

**Review by rossbar — COMMENTED** [review]
I haven't had a chance to take a detailed look yet, but one thing I noticed is that the code in the examples is not formatted properly. If you install the latest pre-commit hooks in your development environment, `black` should automatically run on the documentation examples!

**Comment by dtekinoglu** [comment]
Thank you @rossbar , I believe I did the necessary changes!

File: networkx/algorithms/lowest_common_ancestors.py [hunk 1]
```diff
@@ -35,6 +35,22 @@ def tree_all_pairs_lowest_common_ancestor(G, root=None, pairs=None):
     lcas : generator of tuples `((u, v), lca)` where `u` and `v` are nodes
         in `pairs` and `lca` is their lowest common ancestor.
 
+    Examples
+    --------
+
+    >>> G = nx.DiGraph()
+    >>> G.add_edges_from([(1, 3), (2, 4), (1, 2)])
```
Comment (rossbar) [inline networkx/algorithms/lowest_common_ancestors.py:42]: Just a nit, but an easy way to save a line!

```suggestion
    >>> G = nx.DiGraph([(1, 3), (2, 4), (1, 2)])
```

File: networkx/algorithms/lowest_common_ancestors.py [hunk 2]
```diff
@@ -35,6 +35,22 @@ def tree_all_pairs_lowest_common_ancestor(G, root=None, pairs=None):
...
+    >>> dict(nx.tree_all_pairs_lowest_common_ancestor(G))
+    {(3, 3): 3, (3, 4): 1, (4, 4): 4, (3, 2): 1, (2, 2): 2, (4, 2): 2, (1, 1): 1, (3, 1): 1, (2, 1): 1, (4, 1): 1}
```
Comment (rossbar) [inline networkx/algorithms/lowest_common_ancestors.py:44]: The example is good but unfortunately the resulting dict is not particularly easy to read. Perhaps the results could be printed (or `pprint`ed) instead to have a more readable output?

File: networkx/algorithms/lowest_common_ancestors.py [hunk 3]
```diff
@@ -35,6 +35,22 @@ def tree_all_pairs_lowest_common_ancestor(G, root=None, pairs=None):
...
+
+     We can also use `pairs` argument to specify the pairs of nodes for which we
+     want to compute lowest common ancestors. Here is an example:
```
Comment (rossbar) [inline networkx/algorithms/lowest_common_ancestors.py:47]: It looks like the indentation level here differs from the examples. From a parsing perspective this probably doesn't matter, but from a readability perspective I think it's better if it matches with the examples!

File: networkx/algorithms/lowest_common_ancestors.py [hunk 4]
```diff
@@ -35,6 +35,22 @@ def tree_all_pairs_lowest_common_ancestor(G, root=None, pairs=None):
...
+
+    >>> G = nx.DiGraph()
+    >>> G.add_edges_from([(1, 3), (2, 4), (1, 2)])
```
Comment (rossbar) [inline networkx/algorithms/lowest_common_ancestors.py:50]: There's no need to re-create the sample graph

```suggestion
```

**Review by rossbar — COMMENTED** [review]
These look like nice examples, thanks @dtekinoglu ! Just a few comments/suggestions from a quick readthrough.

**Comment by dtekinoglu** [comment]
@rossbar thanks, I made the suggested changes.

File: networkx/algorithms/lowest_common_ancestors.py [hunk 5]
```diff
@@ -130,6 +152,21 @@ def lowest_common_ancestor(G, node1, node2, default=None):
...
+    >>> nx.lowest_common_ancestor(G, 3, 5)
+    2
+
+    We can also set `default` argument as below. The value of default is returned
+    if there are no common ancestors of given two nodes.
+
+    >>> G = nx.DiGraph()
+    >>> G.add_edges_from([(4, 5), (12, 13)])
+    >>> print(nx.lowest_common_ancestor(G, 12, 5, default="No common ancestors!"))
+    No common ancestors!
```
Comment (rossbar) [inline networkx/algorithms/lowest_common_ancestors.py:168]: This is definitely a nit, but it feels strange to have this wrapped in `print()` as that would really only make sense if you were expecting the output to be a string. This is of course the case for this contrived example, but I think it'd be ever so slightly more clear to remove the print:

```suggestion
    >>> nx.lowest_common_ancestor(G, 12, 5, default="No common ancestors!")
    'No common ancestors!'
```

**Review by rossbar — COMMENTED** [review]
Thanks @dtekinoglu , I think this is almost there - just a few more comments!

**Comment by MridulS** [comment]
I'll go ahead and merge this one @dtekinoglu! Feel free to pick some examples here for the naive algorithms too :)

**Pull request merged**